enhancements/cluster-logging/performance-tuning.md

```diff
@@ -0,0 +1,241 @@
+
+**Note**
+- Only vector will be supported, there are no plans to back-port to fluentd.
+- Existing `output[].limits` allows rate limiting, that is separate from this proposal.
+
+## Motivation
+
+Performance tuning in the underlying collector configuration is complex and error prone.
+Achieving a performance goal may involve configuring multiple parts of the delivery pipeline consistently.
+This would be difficult and error prone for users, but it can be safely automated by our configuration generator.
+
+### User Stories
+#### As a cluster logging administrator I want to minimize log loss.
+
+``` yaml
+outputs:
+ new - name: minimise-log-loss
+   tuning:
+     delivery: safe
+```
+
+#### As a cluster logging administrator I want to maximize throughput.
+
+``` yaml
+outputs:
+  - name: maximise-throughput
+    tuning:
+      delivery: fast
+```
+
+#### As a cluster logging administrator I want to tweak features of a specific output.
+
+``` yaml
+outputs:
+  - name: detailed-tweaks
+    tuning:
+	  delivery: safe
+	  compression: "zlib"  # Enable protocol-specific compression
+	  reconnect:
+	    maxDelay: 1s  # Wait max of 1 second between reconnect attempts
+	  batch:
+		maxBytes: 50K # Send max 50K of data in each write.
+```
+
+### GOALS
+
+- Accomodate users with different performance and reliability trade-offs.
+- Allow users to tune output parameters that we cannot optimize automatically.
+- Expose a simple, general purpose, collector-neutral tuning API
+- Use vector's end-to-end acknowledgements for more efficient log delivery via vector.
+
+### Non-Goals
+
+- Not offering guaruanteed reliable end-to-end delivery (limited by reliability of source and sink)
+- Not exposing underlying vector configuration details.
+- No rate limiting - already provided by `outputs[].limit` field.
+- No plans to support fluentd.
+
+## Proposal
+
+### Workflow Description
+
+**logging administrator** is a human user responsible for setting output tuning parameters as needed.
+
+### API Extensions
+
+New field `ClusterLogForwarder.output[].tuning` with the following sub-fields:
+
+#### delivery: Fast|Safe
+
+`Fast`: Minimize resource use (CPU, disk, network) and maximize throughput in normal operation. Use when:
+- It is acceptable to lose some logs due to restarts, failures or heavy load.
+- The logging system will run close to it's limits, and may sometimes be overloaded.
+- It is important to minimize memory, CPU, disk and network costs.
+
+`Safe`: Minimize log loss due to restarts or failures. Use when:
+- It is important to avoid log loss due to restarts or failures.
+- The logging system is not expected to be pushed close to its limits.
+- Resources (memory, CPU, disk) are sufficient for log throughput to exceed expected peak load.
+
+**Note**: The log collector _cannot_ guarantee reliable end-to-end delivery.
+It has no control over the reliability or throughput of sources and sinks.
+The `delivery` policy tunes the log collector's "best effort".
```

Comment:
To Jeff's point below, I would add a note here pointing to the "implementation details", or maybe specifically mention the following list as "overrides".